cfme/tests/services/test_config_provider_servicecatalogs.py

```diff
@@ -47,22 +51,23 @@ def config_manager(config_manager_obj):
     config_manager_obj.delete()
 
 
-@pytest.fixture(scope="function")
+@pytest.yield_fixture(scope="function")
 def dialog():
-    dialog = "dialog_" + fauxfactory.gen_alphanumeric()
+    dialog_name = "dialog_" + fauxfactory.gen_alphanumeric()
+    service_name = fauxfactory.gen_alphanumeric()
     element_data = dict(
         ele_label="ele_" + fauxfactory.gen_alphanumeric(),
-        ele_name=fauxfactory.gen_alphanumeric(),
+        ele_name="service_name",
         ele_desc="my ele desc",
         choose_type="Text Box",
-        default_text_box="default value"
+        default_text_box=service_name
     )
-    service_dialog = ServiceDialog(label=dialog, description="my dialog",
+    service_dialog = ServiceDialog(label=dialog_name, description="my dialog",
                      submit=True, cancel=True,
                      tab_label="tab_" + fauxfactory.gen_alphanumeric(), tab_desc="my tab desc",
                      box_label="box_" + fauxfactory.gen_alphanumeric(), box_desc="my box desc")
     service_dialog.create(element_data)
-    return dialog
+    yield dialog_name, service_name
 
```

Comment:
Any reason not to clean up after yielding? wrap service_dialog.delete() in a try?